src/NimBLEAttValue.h

```diff
@@ -0,0 +1,210 @@
+/*
+ * NimBLEAttValue.h
+ *
+ *  Created: on March 18, 2021
+ *      Author H2zero
+ *
+ */
+
+#ifndef MAIN_NIMBLEATTVALUE_H_
+#define MAIN_NIMBLEATTVALUE_H_
+#include "nimconfig.h"
+#if defined(CONFIG_BT_ENABLED)
+
+#ifdef NIMBLE_ARDUINO_AVAILABLE
+#include <Arduino.h>
+#endif
+
+#include "NimBLELog.h"
+
+/****  FIX COMPILATION ****/
+#undef min
+#undef max
+/**************************/
+
+#include <string>
+#include <vector>
+#include <time.h>
+
+class NimBLEAttValue
+{
+    uint8_t*     m_attr_value = nullptr;
+    uint16_t     m_attr_max_len = 0;
+    uint16_t     m_attr_len = 0;
+    time_t       m_timestamp = 0;
+
+    void         deepCopy(const NimBLEAttValue & source);
+
+public:
+    NimBLEAttValue(uint16_t init_len = NIMBLE_ATT_INIT_LENGTH, uint16_t max_len = BLE_ATT_ATTR_MAX_LEN);
+    NimBLEAttValue(const uint8_t *value, uint16_t len);
+    NimBLEAttValue(const char *value):NimBLEAttValue((uint8_t*)value, strlen(value)){}
+    NimBLEAttValue(const std::string str):NimBLEAttValue((uint8_t*)str.data(), str.length()){}
+    NimBLEAttValue(const NimBLEAttValue & source) { deepCopy(source); }
```

Comment:
This might crash if `m_attr_value` is not initialized to `nullptr` (check in `deepCopy` will fail)

Suggestion: Initialize `m_attr_value`
```C++
NimBLEAttValue(const NimBLEAttValue & source) : m_attr_value(nullptr) { deepCopy(source); }
```

Reply:
Nice find, noted!